File: kotlin/digital-wallet/src/main/kotlin/digitalwallet/core/domain/models/Wallet.kt
package digitalwallet.core.domain.models

import digitalwallet.core.services.LedgerService
import java.math.BigDecimal
import java.time.LocalDateTime

abstract class Wallet(
    val id: String,
    val customerId: String,
    val policyId: String,
    val insertedAt: LocalDateTime,
) {
    abstract fun getAvailableBalance(ledgerService: LedgerService) : BigDecimal
}

File: scala/digital-wallet/src/main/scala/core/domain/entities/Wallet.scala
package core.domain.entities

import core.domain.enums.WalletType.WalletType

import java.time.LocalDateTime

case class Wallet(
  val id: String,
  val customerId: String,
  val walletType: WalletType,
  val policyId: Option[String],
  val insertedAt: LocalDateTime,
)
